Defend The Divine/Assets/Scripts/GroupSpawner.cs

```diff
@@ -0,0 +1,105 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.VisualScripting;
+using UnityEditorInternal.Profiling.Memory.Experimental.FileFormat;
+using UnityEngine;
+
+/// <summary>
+/// GroupSpawner spawns groups of one enemy type numberOfGroups amount of 
+/// times, then destroys itself.
+/// Implemented this way to allow for more unique waves, e.g. wave 1 spawns a 
+/// group of 4 Enemy1 evry 5 seconds, while also spawning an Enemy2 every
+/// 7 seconds
+/// A "group" is a group of enemies spawned in unison, no matter the size 
+/// (Even if that size is 1)
+/// </summary>
+public class GroupSpawner : MonoBehaviour
+{
+    [SerializeField]
+    GameObject enemyType;
+    [SerializeField]
+    int groupSize;
```

Comment:
it's nice to explicitly declare access like private/public/protected, etc even tho it initializes as private, still nicer to see